pkg/controller/jaeger/jaeger_controller_test.go

```diff
@@ -71,7 +71,7 @@ func TestDeletedInstance(t *testing.T) {
 	// no known objects
 	cl := fake.NewFakeClient()
 
-	r := &ReconcileJaeger{client: cl, scheme: s}
+	r := &ReconcileJaeger{client: cl, scheme: s, rClient: cl}
```

Comment:
Should we use a different `fakeClient` as the `rClient` to ensure that it is invoking the non-cached reader?

Reply:
The `fakeClient` isn't cached, it holds everything in memory already.

Reply:
Okay right. I meant should we ensure its calling the `rClient` and not the `client`

Reply:
Great point. Yes, I think it's a good test to have.